app/services/support_interface/candidate_autofill_usage_export.rb

```diff
@@ -0,0 +1,157 @@
+module SupportInterface
+  class CandidateAutofillUsageExport
+    def data_for_export
+      degree_grade_output +
+        degree_institution_output +
+        degree_subject_output +
+        degree_type_output +
+        other_grade_output +
+        other_type_output
+    end
+
+  private
+
+    def other_type_output
+      counts_for(level: :other, attribute: :qualification_type)
+        .map do |type, count|
+        {
+          'Field' => 'Other type',
+          'Value entered' => type,
+          'Frequency' => count,
+          'Free text?' => !other_types.include?(type),
+        }
+      end
```

Comment:
This block is repeated 6 times. I wonder if we could pull out a method here 

```
    def output(level, attribute, field_name, set_to_check)
      counts_for(level: level, attribute: attribute)
          .map do |attribute, count|
        {
            'Field' => field_name,
            'Value entered' => attribute,
            'Frequency' => count,
            'Free text?' => !set_to_check.include?(attribute),
        }
      end
    end

    def other_type_output
      output(:other, :qualification_type, 'Other type', other_types)
    end

    def other_grade_output
      output(:other, :grade, 'Other grade', other_grades)
    end
    
    ```

Reply:
Yup good idea 👍